runtime/compiler/compile/J9Compilation.cpp

```diff
@@ -855,7 +855,10 @@ J9::Compilation::compileRelocatableCode()
 bool
 J9::Compilation::compilePortableCode()
    {
-   return self()->fej9()->inSnapshotMode();
+   return (self()->fej9()->inSnapshotMode() ||
+	   (compileRelocatableCode() &&
+	    J9_ARE_ANY_BITS_SET(self()->fej9()->getJ9JITConfig()->javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE))
```

Comment:
It's better not to be digging this deep into VM structures this deep into the compiler. What might be cleaner is to add a virtual method in in `TR_J9VMBase` (something like `isPortableSCCEnabled`) that returns `false`, and then in `TR_J9SharedCacheVM`, have the same virtual method return `J9_ARE_ANY_BITS_SET(getJ9JITConfig()->javaVM->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_PORTABLE_SHARED_CACHE)`.

The only real complexity might be that we would need similar queries for JITServer; @mpirvu could you confirm that?